src/uct/ib/rdmacm/rdmacm_iface.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_RDMACM_IFACE_H
+#define UCT_RDMACM_IFACE_H
+
+#include "rdmacm_def.h"
+#include "rdmacm_md.h"
+
+typedef struct uct_rdmacm_iface_config {
+    uct_iface_config_t       super;
+    unsigned                 num_conns;       /** Number of incoming connection */
+    ucs_async_mode_t         async_mode;
+} uct_rdmacm_iface_config_t;
+
+struct uct_rdmacm_iface {
+    uct_base_iface_t          super;
+
+    struct rdma_cm_id         *cm_id;
+    struct rdma_event_channel *event_ch;
+    double                    addr_resolve_timeout;
+
+    uint8_t                   is_server;
+    /** Fields used only for server side */
+    void                                 *conn_request_arg;
+    uct_sockaddr_conn_request_callback_t conn_request_cb;
+
+    uct_rdmacm_ep_t           *ep;
+    ucs_queue_head_t          pending_eps_q;
+
+    struct {
+        ucs_async_mode_t      async_mode;
+        uint32_t              cb_flags;
```

Comment:
should be part of "Fields used only for server side"

Reply:
why? cb_flags are set both for client and server.

Reply:
i think ep has its own cb_flags

Reply:
yes, cb_flags _are_ passed in the ep_create_sockaddr function but since we only support UCT_CB_FLAG_ASYNC now, i saved this value on the iface. both for the client iface and server iface.
shouldn't this value be common though for all the eps on the iface? (keeping in mind that in the future the same iface could be used both for server and client)

Reply:
if the UCT API allows to provide different values, the transport is not allowed to assume they are the same